ui_testing/pages/contact_page.py

```diff
@@ -0,0 +1,454 @@
+            
+        if no_type:
+            return  '-'
+        else:
+            return ', '.join(contact_types)
+
+    def create_update_contact(self, create=True, no='', name='', address='', postalcode='', location='', country='', email='', phone='', skype='', website='', contact_types=['isClient'], departments=[''], contact_persons=True):
+
+        if create:
+            self.base_selenium.LOGGER.info(' + Create new contact.')
+            self.base_selenium.click(element='contacts:new_contact')
+
+            self.base_selenium.LOGGER.info('Wait untill data are loaded')
+            self.sleep_tiny()
+
+        self.set_contact_number(no=no)
+        self.set_contact_name(name=name)
+        self.set_contact_address(address=address)
+        self.set_contact_postalcode(postalcode=postalcode)
+        self.set_contact_location(location=location)
+        self.set_contact_country(country=country)
+        self.set_contact_email(email=email)
+        self.set_contact_phone(phone=phone)
+        self.set_contact_skype(skype=skype)
+        self.set_contact_website(website=website)
+        self.set_contact_type(contact_types=contact_types)
+        contact_departments = self.set_contact_departments(departments=departments)
+
+        self.base_selenium.LOGGER.info('wait to make sure that all data are writtent correctly')
+        self.sleep_tiny()
+        
+        self.base_selenium.LOGGER.info('acquiring contact data')
+        contact_data = self.get_full_contact_data()
+
+        if contact_persons:
+            self.get_contact_persons_page()
+            contact_person_data = self.create_update_contact_person()
+            contact_data['contact_persons'] = contact_person_data
+
+        self.base_selenium.LOGGER.info('Saving the contact created')
+        self.save(save_btn='contact:save')
+        return contact_data
+
+    def get_full_contact_data(self):
+        self.base_selenium.LOGGER.info('Collecting contact data')
+        return {
+            "no": self.get_contact_number(),
+            "name": self.get_contact_name(),
+            "address": self.get_contact_address(),
+            "postalcode": self.get_contact_postalcode(),
+            "location": self.get_contact_location(),
+            "country": self.get_contact_country(),
+            "email": self.get_contact_email(),
+            "phone": self.get_contact_phone(),
+            "skype": self.get_contact_skype(),
+            "website": self.get_contact_website(),
+            "departments": self.get_contact_departments(),
+            "contact_type": self.get_contact_type()
+        }
+
+    def get_contact_persons_page(self):
+        self.base_selenium.LOGGER.info('switch to persons page')
+        self.base_selenium.click(element='contact:contact_persons')
+        self.sleep_tiny()
+
+    def create_update_contact_person(self, create=True, indexToEdit=-1, name='', position='', email='', phone='', skype='', info='', save=False):
```

Comment:
it does both, as if i sent a specific index with specific values, it will set the value of the fields with the new values, so it can be used for both create and update

Reply:
okay